xbmc/windowing/X11/GLContextEGL.h

```diff
@@ -23,6 +23,7 @@
 #if defined(HAVE_X11)
 #include "GLContext.h"
 #include "EGL/egl.h"
+#include <X11/Xutil.h>
```

Comment:
if <X11/Xutil.h> is not included then i get the following build error when compiling:
`In file included from /home/linaro/kodi/xbmc/windowing/X11/GLContextEGL.cpp:37:0:
/home/linaro/kodi/xbmc/windowing/X11/GLContextEGL.h:39:3: error: ‘XVisualInfo’ does not name a type` and subsequent errors

this is related to rockchip as they specifically use ARMs mali GPU that implements OpenGLES (midgard). the corresponding header file regarding egl <EGL/eglplatform.h> of ARM differs from those of other GPU vendors, and also from the mostly used mesa compatible implementations.

in this case the <EGL/egplatform.h> from the usual x86* unix environments implicitly include the mesa compliant implementation of <EGL/platform.h>, and this one automatically includes <X11/Xutil.h> as it assumes X11 automatically for unix. see [https://cgit.freedesktop.org/mesa/mesa/tree/include/EGL/eglplatform.h#n120](url)

ARMs <EGL/egplatform.h> does not automatically include <X11/Xutil.h>. See [https://github.com/rockchip-linux/libmali/blob/rockchip/include/EGL/eglplatform.h](url)

kodi is probably not much used for the platform combination of linux and X11 and ARM GPU OpenGLES.
however kodi probably is very familiar with the combination of android, surfaceflinger and ARMs GPU OpenGLES but then this X11/Xutil.h dependency doesnt exist.

so for rockchip the X11/Xutil.h dependency needs to be specified. so please merge.